### PR TITLE
Add prometheus metrics to monitor replication

### LINKED ISSUE #108: Prometheus metrics
**Describe the feature you'd like to have.**
Scribe should export metrics to a prom endpoint so that monitoring infrastructure can be used to monitor the status of volume replication.

**What is the value to the end user? (why is it a priority?)**
Users/admins need to be able to monitor and receive alerts when data replication fails.

**How will we know we have a good solution? (acceptance criteria)**
- It should be possible to create alerts for failing replication (i.e., the last successful sync was "too long" ago)
- It should be possible to monitor the amount of time the replication iterations are taking (i.e., does it complete well within the sync interval?)

**Additional context**

### PR DESCRIPTION
**Pull request by JohnStrunk**
**Describe what this PR does**
This adds metrics to enable monitoring the state of replication relationships via prometheus

**Is there anything that requires special attention?**

**Related issues:**
Fixes #108

### CODE REVIEW

**Comment by JohnStrunk** [comment]
cc: @mulbc

**Comment by JohnStrunk** [comment]
Metrics example:

Screenshot from 2021-03-12 13-38-46

**Comment by JohnStrunk** [comment]
Metrics example:

Screenshot from 2021-03-12 13-38-46

**Comment by JohnStrunk** [comment]
Metrics example:

Screenshot from 2021-03-12 13-38-46

File: docs/usage/metrics/index.rst [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+or destination):
+
+scribe_missed_intervals_total
+   This is a count of the number of times that a replication iteration failed to
+   complete before the next scheduled start. This metric is only valid for
+   objects that have a schedule (``.spec.trigger.schedule``) specified. For
+   example, when using the rsync mover with a schedule on the source but not on
+   the destination, only the metric for the source side is meaningful.
+scribe_sync_duration_seconds
+   This is a summary of the time required for each sync iteration. My monitoring
```
Comment (mulbc) [inline docs/usage/metrics/index.rst:23]: `By monitoring...`

**Comment by mulbc** [comment]
Could you add an example scrape of your exporter here or (even better) in the documentation? That way it's easier for people knowing Prometheus to know what they should expect...
Also I'll ping my friends in the monitoring ENG to review the monitoring end point to check if they are sane (it's a DB after all and optimizations should happen as early as possible)

**Comment by JohnStrunk** [comment]
I added some raw metric output to the docs. Copied here:
```
$ curl -s http://127.0.0.1:8080/metrics | grep scribe
# HELP scribe_missed_intervals_total The number of times a synchronization failed to complete before the next scheduled start
# TYPE scribe_missed_intervals_total counter
scribe_missed_intervals_total{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 0
scribe_missed_intervals_total{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 0
# HELP scribe_sync_duration_seconds Duration of the synchronization interval in seconds
# TYPE scribe_sync_duration_seconds summary
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.5"} 180.722389767
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.9"} 544.86628289
scribe_sync_duration_seconds{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination",quantile="0.99"} 544.86628289
scribe_sync_duration_seconds_sum{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 1905.815430317
scribe_sync_duration_seconds_count{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 9
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.5"} 9.756509957
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.9"} 19.624571877
scribe_sync_duration_seconds{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source",quantile="0.99"} 19.624571877
scribe_sync_duration_seconds_sum{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 96.962036
scribe_sync_duration_seconds_count{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 9
# HELP scribe_volume_out_of_sync Set to 1 if the volume is not properly synchronized
# TYPE scribe_volume_out_of_sync gauge
scribe_volume_out_of_sync{method="rsync",obj_name="dest",obj_namespace="dstns",role="destination"} 0
scribe_volume_out_of_sync{method="rsync",obj_name="dsrc",obj_namespace="srcns",role="source"} 0
```

What I'm not sure about is what I've chosen to do w/ the `obj_` prefix. When scraped by prometheus, at least in ocp, a namespace label gets automatically added w/ the namespace where the metric was acquired (scribe-system). This overwrites the `namespace` label I had been using to designate where the CR was located--- So I've added the `obj_` prefix to both `name` and `namespace` as they refer to the object that the metric is about.

**Review by husky-parul — APPROVED** [review]

**Pull request merged**